src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -917,10 +919,10 @@ class module_sram : public module_impl
     XRT_PRINTF("<- module_sram::create_instruction_buffer()\n");
   }
 
-  void
-  patch_instr(const std::string& argnm, const xrt::bo& bo) override
+  virtual void
+  patch_instr(const std::string& argnm, const xrt::bo& bo, const patcher::buf_type type) override
```

Comment:
No typical adding const on by value paramters.  
```suggestion
  patch_instr(const std::string& argnm, const xrt::bo& bo, patcher::buf_type type) override
```